aggregations/db_tables/daily_new_unique_contracts_count.py

```diff
@@ -46,33 +41,24 @@ def sql_insert(self):
         """
 
     def collect(self, requested_timestamp: int) -> list:
-        # Get new contracts from Indexer DB. We use `distinct` in SQL,
-        # But we still have no guarantees because the contract could be added a week ago
-        new_contracts = super().collect(requested_timestamp)
-
-        all_contract_hashes_select = """
+        new_unique_hashes_select = """
             SELECT code_sha256
             FROM deployed_contracts
-            WHERE first_created_by_block_timestamp < %(timestamp)s
+            WHERE first_created_by_block_timestamp >= %(from_timestamp)s
+                AND first_created_by_block_timestamp < %(to_timestamp)s
         """
 
         from_timestamp = self.start_of_range(requested_timestamp)
+        computed_for_date = datetime.datetime.utcfromtimestamp(from_timestamp).strftime(
+            "%Y-%m-%d"
+        )
         with self.analytics_connection.cursor() as analytics_cursor:
-            # Get all contracts that were added before our time range
             analytics_cursor.execute(
-                all_contract_hashes_select, time_json(from_timestamp)
-            )
-            previous_contracts = analytics_cursor.fetchall()
-
-            # Find truly unique contracts
-            new_unique_contracts = [
-                c for c in new_contracts if c not in previous_contracts
-            ]
-
-            time = datetime.datetime.utcfromtimestamp(from_timestamp).strftime(
-                "%Y-%m-%d"
+                new_unique_hashes_select,
+                time_range_json(from_timestamp, self.duration_seconds),
             )
-            return [(time, len(new_unique_contracts))]
+            new_unique_contracts = analytics_cursor.fetchall()
+            return [(computed_for_date, len(new_unique_contracts))]
```

Comment:
We don't need `len()`, instead we can query the `COUNT()` directly from the database.